PriceCalculatorKata/Product.cs

```diff
@@ -1,13 +1,22 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Text;
+using PriceCalculatorKata.Report;
 
 namespace PriceCalculatorKata
 {
     public class Product
     {
-        public static float discount = 0;
         public string Name { get; set; }
+        private IPriceCalculator _priceCalculator;
+        private IReporter _reporter;
+        public Product(int UPC,float basePrice)
+        {
+            this.UPC = UPC;
+            this.BasePrice = basePrice;
+            _priceCalculator = new ProductPriceCalculator(this);
```

Comment:
I don't feel it's wise to have the ProductPriceCalculator and ProductReporter depend on the product ( the product being injected to these classes constructors) 
ProductPriceCalculator and ProductReporter encapsulate a shared behavior between products so we can just pass the product to there Calculate, Display methods